x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/OrdinalsGroupingOperator.java

```diff
@@ -208,25 +237,28 @@ public Page getOutput() {
         if (finished == false) {
             return null;
         }
+        Page page = null;
         if (valuesAggregator != null) {
             try {
-                return valuesAggregator.getOutput();
+                page = valuesAggregator.getOutput();
             } finally {
                 final ValuesAggregator aggregator = this.valuesAggregator;
                 this.valuesAggregator = null;
                 Releasables.close(aggregator);
             }
-        }
-        if (ordinalAggregators.isEmpty() == false) {
+        } else if (ordinalAggregators.isEmpty() == false) {
             try {
-                return mergeOrdinalsSegmentResults();
+                page = mergeOrdinalsSegmentResults();
```

Comment:
And this one?

That way in the `profile` output we can tell which of the two we used.